examples/advanced/edge/jobs/cifar10_mobile_et/app/custom/evaluator.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import torch
+import torchvision
+import torchvision.transforms as transforms
+from torch.utils.tensorboard import SummaryWriter
+
+from nvflare.apis.fl_constant import FLContextKey
+from nvflare.apis.fl_context import FLContext
+from nvflare.app_common.abstract.model import ModelLearnableKey
+from nvflare.app_common.app_constant import AppConstants
+from nvflare.app_common.app_event_type import AppEventType
+from nvflare.edge.models.model import Cifar10ConvNet
+from nvflare.widgets.widget import Widget
+
+DEVICE = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+class GlobalEvaluator(Widget):
+    def __init__(self, data_root: str):
+        super().__init__()
+        self.model = Cifar10ConvNet()
+        self.data_root = data_root
+        self.batch_size = 4
+        self.test_loader = None
+        self.tb_writer = None
+
+        self.register_event_handler(AppEventType.AFTER_SHAREABLE_TO_LEARNABLE, self.evaluate)
+        self.register_event_handler(AppEventType.TRAINING_STARTED, self.initiate_tb)
```

Comment:
Should change to use EventType.START_RUN. In theory there could be multiple "sage" controllers, and each could fire a TRAINING_STARTED event, which will cause the initiate_cb to be called multiple times.
START_RUN event is guaranteed to be fired once for the whole job.

Reply:
updated